pysat/_instrument.py

```diff
@@ -1497,46 +1527,156 @@ def load(self, yr=None, doy=None, date=None, fname=None, fid=None,
         are automatically applied to the data before it is available to
         user in .data.
 
+        A mixed combination of `.load()` keywords such as `yr` and `date` are
+        not allowed.
+
+        Examples
+        --------
+        ::
+            import datetime as dt
+            import pysat
+
+            inst = pysat.Instrument('pysat', 'testing')
+
+            # load a single day by year and day of year
+            inst.load(2009, 1)
+
+            # load a single day by date
+            date = dt.datetime(2009, 1, 1)
+            inst.load(date=date)
+
+            # load a single file, first file in this example
+            inst.load(fname=inst.files[0])
+
+            # load a range of days, data between
+            # Jan. 1st (inclusive) - Jan. 3rd (exclusive)
+            inst.load(2009, 1, 2009, 3)
+
+            # same procedure using datetimes
+            date = dt.datetime(2009, 1, 1)
+            end_date = dt.datetime(2009, 1, 3)
+            inst.load(date=date, end_date=end_date)
+
+            # same procedure using filenames
+            # note the change in index due to inclusive slicing on filenames!
+            inst.load(fname=inst.files[0], stop_fname=inst.files[1])
+
         """
-        # set options used by loading routine based upon user input
-        if date is not None:
-            # ensure date portion from user is only year, month, day
-            self._set_load_parameters(date=date, fid=None)
 
-            # increment by a day
-            inc = pds.DateOffset(days=1)
-            curr = self._filter_datetime_input(date)
-        elif (yr is not None) & (doy is not None):
+        # set options used by loading routine based upon user input
+        if (yr is not None) & (doy is not None):
+            # verify arguments make sense, in context
+            _check_load_arguments_none(fname, stop_fname, date, end_date,
+                                       raise_error=True)
+            # convert yr/doy to a date
             date = dt.datetime(yr, 1, 1) + pds.DateOffset(days=(doy - 1))
```

Comment:
```suggestion
            date = dt.datetime.strptime("{:.0f} {:.0f}".format(yr, doy), "%Y %j")
```

Making this suggestion because it's faster and simpler than the current method.  Using timeit, the suggestion took 0.2198226160253398 s while the current method took 0.7341184029937722s.